### DISCUSSION TITLE
RipsLayer

### DISCUSSION THREAD

**Discussion by fbarfi** [GUDHI/gudhi-devel#965] [Q&A]
Would it make a difference in terms of computational speed of the diagrams to use buffer_size in RipsLayer?
Thanks.

**Comment by mglisse** [comment]
I am not very familiar with tensorflow (@MathieuCarriere handles that), but I doubt it would make a difference, what takes time is really the C++ code to construct a Rips complex and compute its persistent homology on the CPU. The current plan to improve performance is
- tweak edge-collapse so it preserves the information necessary for derivation
- use a modified version of ripser

It would still likely be the bottleneck of whatever network it is included in, but it should gain quite a bit. However neither will be for the next release.

Out of curiosity, in what ambient dimension are you using RipsLayer? We also have plans at some point to add a CechLayer (actually using alpha-complex or Delaunay-Cech), which would be faster if the ambient dimension is low. That's for even further in the future though.

**Reply by fbarfi** [reply]
@mglisse thank you for your prompt response. I am using homology =1 . Do you think that using alpha complex might speed up the process of calculating the diagrams? I am calculating the diagrams for a sociomatrix with thousands of nodes. best.

**Reply by mglisse** [reply]
I was asking about the **ambient** dimension (number of coordinates of the points), not the dimension of the homology groups. If the ambient dimension is low (say 3 for instance), then the alpha complex would be much faster to compute than the Rips. However, we do not currently have a differentiable version in Gudhi, so that wouldn't help you. If your points are not even in a Euclidean space, you can forget I even mentioned the alpha-complex.

**Reply by fbarfi** [reply]
thanks. I have an ambient space.with high dimensions, that is why it is taking a long time to compute the diagrams.
